Automattic/wp-calypso
# Signup: remove redundant removeUnneededSteps

#### Changes proposed in this Pull Request

As a follow up to #35049, this PR cleans up all references to the `removeUnneededSteps` action and the `SIGNUP_PROGRESS_REMOVE_UNNEEDED_STEPS` action type.

#### Testing instructions

As these changes are purely janitorial there should be no change to sign up behaviour so please test a number of scenarios in signup - different flows, different directions (use the back button a few of times from the last step etc). 
That said, a particular area to watch out for here is in switching flows, so let's make sure that that's safe with the following:

- Starting at `http://calypso.localhost:3000/start/premium`
- Choose blog and continue on through the steps until you reach `start/premium/site-title-with-preview`
  - note that you're still on the `premium` flow (`/start/premium`)
- Go back to the site-type step
- Select "Online Store" as your site type.
  - Note that you're now the `ecommerce-onboarding` flow (`start/ecommerce-onboarding`)
- Go back again
  - note that you're back on the `premium` flow (`/start/premium`)
- Overall, nothing there should be too surprising - no BSoD's, no oddities with the progress counts, no missing steps etc.
